src/xkbcomp/expr.c

```diff
@@ -656,11 +656,25 @@ ExprResolveKeySym(struct xkb_context *ctx, const ExprDef *expr,
     if (!ExprResolveInteger(ctx, expr, &val))
         return false;
 
-    if (val < 0 || val >= 10)
```

Comment:
This looks like it regressed in 2e4933cda7c141c9efc8c62e7d72f591f2998836 compared to xkbcomp...